File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
      
        Stack<Integer> st = new Stack<>();
        for (int a : arr){
            if (st.empty()){
                st.push(a);
            }
            else if (st.peek()!=a){
                st.push(a);
            }
        }
        
        int[] answer = new int[st.size()];
        for (int i = st.size()-1; i>=0; i--){
            answer[i] = st.pop();
        }
        
        return answer;
    }
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    stack = []
    for a in arr:
        if not stack:  # 스택이 비어있는 경우
            stack.append(a)
        elif stack[-1] != a:  # 스택의 마지막 요소와 현재 요소가 다른 경우
            stack.append(a)
    
    # 스택에 남은 요소가 정답
    return stack